packages/app/src/components/Wire.tsx

```diff
@@ -49,6 +51,22 @@ export const RenderedWire: FC<RenderedWireProps> = ({
     return null;
   }
 
+  const { inputDefs: fromInputDefs, outputDefs: fromOutputDefs } = useNodeIO(connection.fromNodeId);
+  const { inputDefs: toInputDefs, outputDefs: toOutputDefs } = useNodeIO(connection.toNodeId);
+
+  // Compute the color of the wire based on the connected ports' datatypes
+  const wireColor = useMemo(() => {
+    const fromPortDef = fromOutputDefs.find(def => def.name === connection.fromPortName) ||
+                        fromInputDefs.find(def => def.name === connection.fromPortName);
+    const toPortDef = toInputDefs.find(def => def.name === connection.toPortName) ||
+                      toOutputDefs.find(def => def.name === connection.toPortName);
+
+    const fromDataType = Array.isArray(fromPortDef?.type) ? fromPortDef.type[0] : fromPortDef?.type || 'unknown';
+    const toDataType = Array.isArray(toPortDef?.type) ? toPortDef.type[0] : toPortDef?.type || 'unknown';
```

Comment:
A shared data types should be considered between fromData and toData, but here you only consider the first data type in each port data